traefik/templates/_helpers.tpl

```diff
@@ -92,6 +92,32 @@ are multiple namespaced releases with the same release name.
 {{- (printf "%s-%s" (include "traefik.fullname" .) (include "traefik.namespace" .)) | trunc 63 | trimSuffix "-" }}
 {{- end -}}
 
+{{/*
+Change input to a valid name for a port.
+This is a best effort to convert input to a valid port name,
+which only allows lowercase alphanumeric characters and '-',
+and imposes limits on the length of the name.
+See also https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.32/#containerport-v1-core.
+*/}}
+{{- define "traefik.portname" -}}
+{{- $portName := . -}}
+{{- $portName = $portName | lower -}}
```

Comment:
upper case seems to be valid, according to the RFC.

Reply:
See comment above